### DISCUSSION TITLE
Port Forwarding by Xray-core

### DISCUSSION THREAD

**Discussion by fodhelper** [XTLS/Xray-core#4141] [Ideas]
Hey
What do you think about Port Forwarding function for Xray-core?
As SSH Port Forwarding and SSH Reverse Tunnel is Blocked or Will cause server getting blocked by GFW after usage
It's a good feature if Xray-core support it

By being able to forwarding a port from server to client, client can access for example a MySQL service running on the server

And by being able to forward a port from client to server, you can expose a local port to public
Or you can do a reverse proxy by making an XHTTP proxy at Chinese server(that have a Public IP), and using an American server as a client, and forwarding a port from client to server, the use the Chinese server as relay to connect to the American server or even any other IP

I know there is already other tools to do similar things, but none of them have exclusive transports of Xray like XHTTP and XMUX, and their performance is also not as good as Xray

**Comment by RPRX** [comment]
试试客户端任意门进，服务端 freedom 重定向

**Reply by fodhelper** [reply]
Could you please share samples?
Port 443 of server is the proxy port(VLESS XHTTP TLS), I want to forward incoming traffics of server(port 8443) to the client(port 443)

**Reply by AmirReza2012** [reply]
@RPRX Hy2 has a feature called [tcpForwarding](https://v2.hysteria.network/docs/advanced/Full-Client-Config/#:~:text=realm%3A%20martian-,TCP%20Forwarding,-TCP%20Forwarding%20allows) (and udpForwarding) where it forwards a port from the server to the client **through** the proxy connection.

Is it possible on xray to make Doko-demo door go through a proxy outbound?

**Comment by shakibamoshiri** [comment]
you can use dokodemo-door acting as port-forwarding over any proxy you have , could be forward or tunnel 
https://xtls.github.io/config/inbounds/tunnel.html#tunnel-dokodemo-door

first setup you proxy and make sure it is working , the try

```json
    {
      "listen": "127.0.0.1",
      "port": 10321,
      "protocol": "dokodemo-door",
      "tag": "inbound-127.0.0.1:10321",
      "settings": {
        "address": "127.0.0.1",
        "port": 10321,
        "network": "tcp,udp",
        "followRedirect": false,
        "timeout": 0
      },
      "streamSettings": null,
      "sniffing": {
        "enabled": false,
        "destOverride": [
          "http",
          "tls",
          "quic",
          "fakedns"
        ],
        "metadataOnly": false,
        "routeOnly": false
      }
    },

```

this will forward any request on port 10123 to other side of tunnel port 10123

this is what you need to know 
https://www.youtube.com/watch?v=hzlLC_GBRcI&list=PL_akDzx9nCAEYAZByaDGVJM5Sw_2HWoF9&index=12&pp=gAQBiAQB
